libndt7/src/main/java/net/measurementlab/ndt7/android/Client.kt

```diff
@@ -0,0 +1,184 @@
+package net.measurementlab.ndt7.android
+
+import android.support.annotation.CheckResult
+import android.util.Log
+import com.google.gson.Gson
+
+import org.json.JSONException
+import org.json.JSONObject
+
+import java.net.URI
+import java.net.URISyntaxException
+import java.security.cert.X509Certificate
+import java.util.concurrent.TimeUnit
+
+import javax.net.ssl.SSLContext
+import javax.net.ssl.TrustManager
+import javax.net.ssl.X509TrustManager
+
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.WebSocket
+import okhttp3.WebSocketListener
+import okio.ByteString
+
+// TODO(bassosimone): do we need locking for this class?
+
+private const val TAG = "Client"
+
+open class Client(private val settings: Settings) : WebSocketListener() {
+
+    private val gson = Gson()
+    private val measurementInterval = TimeUnit.NANOSECONDS.convert(250, TimeUnit.MILLISECONDS)
+
+    private var count = 0.0
+    private var rv = true
+    private var t0: Long = 0
+    private var tLast: Long = 0
+
+    open fun onLogInfo(message: String?) {
+        Log.d(TAG, "onLogInfo: " + message!!)
```

Comment:
What is the meaning of `message!!`?

Reply:
In Kotlin that means assume it's not null.  But can remove it and replace with string interpolation.